WooCommerce/src/androidTest/kotlin/com/woocommerce/android/screenshots/orders/UnifiedOrderScreen.kt

```diff
@@ -0,0 +1,116 @@
+package com.woocommerce.android.screenshots.orders
+
+import androidx.test.espresso.Espresso
+import androidx.test.espresso.action.ViewActions
+import androidx.test.espresso.action.ViewActions.click
+import androidx.test.espresso.assertion.ViewAssertions.matches
+import androidx.test.espresso.matcher.ViewMatchers
+import androidx.test.espresso.matcher.ViewMatchers.hasDescendant
+import androidx.test.espresso.matcher.ViewMatchers.isDisplayed
+import androidx.test.espresso.matcher.ViewMatchers.withEffectiveVisibility
+import androidx.test.espresso.matcher.ViewMatchers.withId
+import androidx.test.espresso.matcher.ViewMatchers.withText
+import com.woocommerce.android.R
+import com.woocommerce.android.screenshots.util.NestedScrollViewExtension
+import com.woocommerce.android.screenshots.util.Screen
+import org.hamcrest.core.AllOf.allOf
+
+class UnifiedOrderScreen : Screen(ORDER_CREATION) {
+    companion object {
+        const val CREATE_BUTTON = R.id.menu_create
+        const val CUSTOMER_SECTION = R.id.customer_section
+        const val DONE_BUTTON = R.id.menu_done
+        const val EDIT_STATUS_BUTTON = R.id.orderStatus_editButton
+        const val FEE_BUTTON = R.id.fee_button
+        const val ORDER_CREATION = R.id.order_creation_root
+        const val PAYMENT_SECTION = R.id.payment_section
+        const val PRODUCTS_SECTION = R.id.products_section
+        const val SHIPPING_BUTTON = R.id.shipping_button
+        const val TOOLBAR = R.id.collapsing_toolbar
+        const val UPDATE_STATUS_LIST_VIEW = R.id.select_dialog_listview
+    }
+
+    fun createOrder(): SingleOrderScreen {
+        clickOn(CREATE_BUTTON)
+        return SingleOrderScreen()
+    }
+
+    fun updateOrderStatus(newOrderStatus: String): UnifiedOrderScreen {
+        clickOn(EDIT_STATUS_BUTTON)
+        waitForElementToBeDisplayed(UPDATE_STATUS_LIST_VIEW)
+        Espresso.onView(withText(newOrderStatus))
+            .perform(click())
+        Espresso.onView(withText("Apply"))
+            .perform(click())
+        return this
+    }
+
+    fun clickAddCustomerDetails(): CustomerDetailsScreen {
+        waitForElementToBeDisplayed(PAYMENT_SECTION)
+        Espresso.onView(withId(CUSTOMER_SECTION))
+            .perform(NestedScrollViewExtension())
+        Espresso.onView(withText("Add customer details"))
+            .perform(click())
+
+        waitForElementToBeDisplayed(
+            Espresso.onView(
+                allOf(
+                    ViewMatchers.isDescendantOfA(withId(CustomerDetailsScreen.FIRST_NAME_INPUT)),
+                    ViewMatchers.withHint(CustomerDetailsScreen.FIRST_NAME_INPUT_HINT_TEXT),
+                    ViewMatchers.isCompletelyDisplayed()
+                )
+            )
+        )
```

Comment:
1. We don't have to use this simplistic wait for first name element to appear any more, since we now have `waitForElementToDisappear(PROGRESS_BAR)`
2. WDYT of moving the `waitForElementToDisappear(PROGRESS_BAR)` from `addCustomerDetails` here? It makes sense to me, because we need the progress bar to disappear after we open the screen, in every case we open this screen. It happens that the only thing we do in `CustomerDetailsScreen` currently is entering customer details, but we might extend this screen with more methods - and in such case we'll need to have this part in every method. So I think it makes sense to have this waiter be a part of `screen opening` sequence.

```suggestion
        waitForElementToDisappear(PROGRESS_BAR)
```

Reply:
Yup, I missed this and agree that we don't need it as part of `clickAddCustomerDetails` anymore. Updated in https://github.com/woocommerce/woocommerce-android/pull/7443/commits/86ac233e903023d7514cda079172f6ea2a79693d 

However I would prefer to keep `waitForElementToDisappear(PROGRESS_BAR)` as part of `CustomerDetailsScreen` so we don't have to call an element from a separate screen in another screen (`PROGRESS_BAR` is an element on `CustomerDetailsScreen`). 

When we do extend `CustomerDetailsScreen` with more methods, I think we could create a new "flow" folder where functions created on this can be a combination of methods from different screens. To keep the screen clean (so that it's easier to maintain/delete screens in the future), I think it's better to only use elements that are on the screen, if there's a need to interact with elements on multiple screens, I think it should be on the test or flow level.